File: pgm/ABC_Dart/ABC_398_A.dart
// import 'dart:collection';
import 'dart:io';
// import 'dart:math';

void main() {
  final int N = int.parse(stdin.readLineSync()!);
  final String hyphen = "-";
  final String equal = "=";
  final int two = 2;
  final String lr = hyphen * ((N - two + N % two) ~/ two);
  final String c = equal * (two - N % two);
  print("$lr$c$lr");
  exit(0);
}

File: pgm/ABC_PHP/ABC_398_A.php
<?php
$stream = fopen('php://stdin', 'r');
$line = fgets($stream);
fclose($stream);
const equal = '=';
const hyphen = '-';
const two = 2;
$N = (int) trim($line);
$lr = str_repeat(hyphen, intdiv($N - two + $N % two, two));
$c = str_repeat(equal, two - $N % two);
echo $lr . $c . $lr . PHP_EOL;
exit(0);
